cli/src/pcluster/templates/cw_dashboard_builder.py

```diff
@@ -219,6 +228,131 @@ def _add_conditional_storage_widgets(
                 widgets_list.append(graph_widget)
         return widgets_list
 
+    def custom_pcluster_metric_filter(self, metric_name, filter_pattern):
+        """Adding custom metric filter from named tuple."""
+        metric_filter = logs.CfnMetricFilter(
+            scope=self.stack_scope,
+            id=metric_name + " Filter",
+            filter_pattern=filter_pattern,
+            log_group_name=self.cw_log_group_name,
+            metric_transformations=[
+                logs.CfnMetricFilter.MetricTransformationProperty(
+                    metric_namespace="ParallelCluster/Errors",
+                    metric_name=metric_name,
+                    metric_value="1",
+                    default_value=0,
+                )
+            ],
+        )
+        metric_filter.add_depends_on(self.cw_log_group)
+        return metric_filter
+
+    def add_custom_error_metrics(self):
+        """Create custom error metric filter and outputs to cloudwatch graph."""
+        jobs_not_starting_errors = [
+            custom_metric_filter(
+                error_information="Missing IAM Permissions",
+                metric_name="No IAM Group",
+                filter_pattern="UnauthorizedOperation",
+            ),
+            custom_metric_filter(
+                error_information="Missing IAM Permissions",
+                metric_name="Mismatch between IAM Group",
+                filter_pattern="AccessDeniedException",
+            ),
```

Comment:
What is the difference between the two metrics? How do I reproduce each of them?

Please consider merging them:
```
custom_metric_filter(
                error_information="Missing IAM Permissions",
                metric_name="Missing IAM Permissions",
                filter_pattern="UnauthorizedOperation|AccessDeniedException",
            ),
```

Reply:
techniocally its different, the above one is if the head node doesn't have an IAM group, the bottom one is if you assign it to a different cluster's IAM